app/forms/openid_connect_authorize_form.rb

```diff
@@ -65,6 +65,7 @@ def verified_at_requested?
   end
 
   def service_provider
+    return NullServiceProvider.new(issuer: nil) if client_id.include?("\x00")
```

Comment:
I was hoping to use `validate_client_id`, but `service_provider` gets called in `#initialize`, so the error occurs before any validations happen.